pennylane/beta/tapes/tape.py

```diff
@@ -936,3 +936,387 @@ def execute_device(self, params, device):
+          where ``n = tape.num_params``.
+
+        * Analytic method (``'analytic'``). Analytic, if implemented by the inheriting quantum tape.
+
+        * Best known method for each parameter (``'best'``): uses the analytic method if
+          possible, otherwise finite difference.
+
+        * Device method (``'device'``): Delegates the computation of the Jacobian to the
+          device executing the circuit. Only supported by devices that provide their
+          own method for computing derivatives; support can be checked by
+          querying the device capabilities: ``dev.capabilities()['provides_jacobian']`` must
+          return ``True``. Examples of supported devices include the experimental
+          :class:`"default.tensor.tf" <~.DefaultTensorTF>` device.
+
+        .. note::
+
+            The finite difference method is sensitive to statistical noise in the circuit output,
+            since it compares the output at two points infinitesimally close to each other. Hence
+            the ``'F'`` method works best with exact expectation values when using simulator
+            devices.
+
+        Args:
+            device (~.Device, ~.QubitDevice): a PennyLane device
+                that can execute quantum operations and return measurement statistics
+            params (list[Any]): The quantum tape operation parameters. If not provided,
+                the current tape parameter values are used (via :meth:`~.get_parameters`).
+
+        Keyword Args:
+            method="best" (str): The differentiation method. Must be one of ``"numeric"``,
+                ``"analytic"``, ``"best"``, or ``"device"``.
+            h=1e-7 (float): finite difference method step size
+            order=1 (int): The order of the finite difference method to use. ``1`` corresponds
+                to forward finite differences, ``2`` to centered finite differences.
+
+        Returns:
+            array[float]: 2-dimensional array of shape ``(tape.num_params, tape.output_dim)``
+
+        **Example**
+
+        .. code-block:: python
+
+            from pennylane.beta.tapes import QuantumTape
+            from pennylane.beta.queuing import expval, var, sample, probs
+
+            with QuantumTape() as tape:
+                qml.RX(0.432, wires=0)
+                qml.RY(0.543, wires=0)
+                qml.CNOT(wires=[0, 'a'])
+                qml.RX(0.133, wires='a')
+                probs(wires=[0, 'a'])
+
+        If parameters are not provided, the existing tape parameters are used:
+
+        >>> dev = qml.device("default.qubit", wires=[0, 'a'])
+        >>> tape.jacobian(dev)
+        array([[-0.178441  , -0.23358253, -0.05892804],
+               [-0.00079144, -0.00103601,  0.05892804],
+               [ 0.00079144,  0.00103601,  0.00737611],
+               [ 0.178441  ,  0.23358253, -0.00737611]])
+
+        Parameters can be optionally passed during execution:
+
+        >>> tape.jacobian(dev, params=[1.0, 0.0, 1.0])
```

Comment:
Would this mean that the passed `params` list must be the same length as the number of parameters for the gates in the tape? What would happen if using nested tapes?

Reply:
To determine the tape parameters, what the tape does is loop through all objects, and accumulate the output of `object.data`.

* For operations, this is simply the raw parameter list.

* For tapes, this is an alias to `tape.get_parameters(trainable_only=False)` 🙂 

So it should work fine with nested tapes (and I believe there is a pre-existing test that checks this in `test_tape.py` from an earlier PR)